### PR TITLE
Use `AZURE_QUANTUM_CONNECTION_STRING` for authentication

### PR DESCRIPTION
**Pull request by cqc-alec**
This works on the CI.

Also add a timeout parameter to `get_result()`; and make config methods available in the package.

### CODE REVIEW

**Comment by cqc-alec** [comment]
One test is now failing because of a timeout, but I think we should merge this as it gets the CI working properly.

File: tests/backend_test.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def test_quantinuum_option_params(azure_backend: AzureBackend) -> None:
     c1 = b.get_compiled_circuit(c)
     if b.is_available() and b.average_queue_time_s() < 600:
         h = b.process_circuit(c1, n_shots=1000, option_params={"error_model": False})
-        r = b.get_result(h)
+        r = b.get_result(h, timeout=1200)
```
Comment (cqc-melf) [inline tests/backend_test.py:64]: Do you want this timeout to be 1200 ? The other are 120

**Comment by cqc-melf** [comment]
Do we want to update the CI to test other os, too?

File: tests/backend_test.py (hunk 1 above)
Comment (cqc-alec) [reply tests/backend_test.py:64]: Yes, I set it to them all to double the max reported average queue time (see three lines above).

**Comment by cqc-alec** [comment]
> Do we want to update the CI to test other os, too?

For now I think one OS is enough; all the tests use credits and there's very little value in testing on different OSs.

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**